### PR TITLE
Get rid of DoubleHelper in Number.FormatAndParse.cs

### PR DESCRIPTION
**Pull request by EgorBo**
See https://github.com/dotnet/coreclr/pull/18853
(makes code smaller and BigEndian friendly)

### CODE REVIEW

File: src/System.Memory/src/System/Number/Number.FormatAndParse.cs [hunk 1]
```diff
@@ -25,22 +25,13 @@ internal static partial class Number
...
             }
 
-            if (e == 0 && m == 0)
-            {
-                d = 0;
-            }
-
-            value = d;
-
+            value = (d == 0.0) ? 0.0 : d;
```
Comment (tannergooding) [inline src/System.Memory/src/System/Number/Number.FormatAndParse.cs]: A comment indicating this is normalizing `-0.0` to `+0.0` would be nice 😄

**Comment by jkotas** [comment]
The NetFX build break should be fixed once #30897 gets merged.

**Comment by ViktorHofer** [comment]
@dotnet-bot test this please

**Comment by ViktorHofer** [comment]
@dotnet-bot test OSX x64 Debug Build

**Pull request merged**